sstables/mx/writer.cc

```diff
@@ -1003,6 +1004,7 @@ void writer::consume(tombstone t) {
 
     if (t) {
         _collector.update_min_max_components(clustering_key_prefix::make_empty(_schema));
+        _cannot_have_partition_tombstones = false;
```

Comment:
@kbr- could u pls explain to me why we don't need to track range tombstones as well? as part of https://github.com/scylladb/scylla/pull/7437 work. would we have a problem in 7437 if we have a range tombstone which is essentially a partition tombstone?

Reply:
In this commit https://github.com/scylladb/scylla/pull/7437/commits/34bb4d6da434d8cf55e8932c137da3466dc44abb, I found this comment:
    // 3. The sstables cannot have partition tombstones for the same reason as above. There
    //    is no way to check exactly this, but there is a way to check if the sstable has no tombstones
    //    of any type, so we use that. TWCS sstables will usually pass this condition.

But we aren't checking for range tombstones here, and I think they're important when doing the single-partition query filtering optimization.

Reply:
The comment is outdated. It turned out that we don't have a way that "sstable has no tombstones of any type", hence this PR.

Range tombstones are OK. Only partition tombstones are a problem since the new reader merger could encounter a partition tombstone in the middle of a partition query long after it returned a `partition_start` fragment.

Reply:
Range tombstones have a position in partition and are returned in order of their positions.

Partition tombstones are always returned as the first thing by mutation readers (in the `partition_start` fragment). However, the reader merger from #7437 is opening new mutation readers in the middle of a partition, after it already merged fragments returned from other readers. So imagine the following scenario: we're merging readers `r1` and `r2`:
1. the merger opens `r1`
2. `r1` returns `partition_start` with no tombstone, the merger returns it
3. `r1` returns a clustering row, the merger returns it
4. the merger opens `r2`
5. `r2` returns `partition_start` with a partition tombstone
6. ??? The merger cannot return it since it's illegal to return a `partition_start` twice. And it cannot merge it with the first `partition_start` since it already returned it.

The simplest way to deal with this problem is to ensure that there are no partition tombstones.

Reply:
thanks for the explanation!